mod/cli/pkg/commands/deposit/create.go

```diff
@@ -139,55 +151,167 @@ func createValidatorCmd(
 			"signature", signature.String(),
 		)
 
-		// TODO: once broadcast is fixed, remove this.
-		logger.Info("Send the above calldata to the deposit contract 🫡")
+		if broadcast {
+			var txHash common.ExecutionHash
+			txHash, err = broadcastDepositTx[ExecutionPayloadT](
+				cmd, depositMsg, signature, chainSpec,
+			)
+			if err != nil {
+				return err
+			}
+
+			logger.Info(
+				"Deposit transaction successful",
+				"txHash", txHash.Hex(),
+			)
+		}
 
 		return nil
 	}
 }
 
+func broadcastDepositTx[
+	ExecutionPayloadT constraints.EngineType[ExecutionPayloadT],
+](
+	cmd *cobra.Command,
+	depositMsg *types.DepositMessage,
+	signature crypto.BLSSignature,
+	chainSpec common.ChainSpec,
+) (common.ExecutionHash, error) {
+	// Parse the private key.
+	var (
+		fundingPrivKey string
+		privKey        *ecdsa.PrivateKey
+	)
+	fundingPrivKey, err := cmd.Flags().GetString(privateKey)
+	if err != nil {
+		return common.ExecutionHash{}, err
+	}
+	if fundingPrivKey == "" {
```

Comment:
probably one of these checks is not required since "" => nil pk i believe

Reply:
I removed above check, and on testing by not passing privateKey and setting broadcast flag as true, getting error `2024/07/15 22:18:43 ERROR startup failure error="invalid length, need 256 bits"` which is not giving clarity I believe. This check specifically tells users that the private key is required.